lib/roby/app.rb

```diff
@@ -3048,16 +3067,15 @@ def each_test_file_in_app
                     break
                 end
             end
+        end
 
-            Find.find(dir) do |path|
-                # Skip the robot-specific bits that don't apply on the
-                # selected robot
-                Find.prune if File.directory?(path) && !autodiscover_tests_in?(path)
+        # Enumerate all the test files in this app and for this robot
+        # configuration
+        def each_test_file_in_app
+            return enum_for(__method__) unless block_given?
 
-                if File.file?(path) && /^test_.*\.rb$/.match?(File.basename(path))
-                    yield(path)
-                end
-            end
+            dir = File.join(app_dir, "test")
+            each_test_file_in(dir)
```

Comment:
Since it is a `each` method, you need to forward the block to `each_test_file_in_dir`

~~~
def each_test_file_in_app(&block)
   return enum_for(...) unless block_given?
   
   each_test_file_in(dir, &block)
end
~~~

Now, since `each_test_file_in_dir` already handles the no-block situation, you can also forget the `return enum_for...` bit. 

~~~
def each_test...(&block)
   dir =
   each_test_file_in(dir, &block)
end
~~~